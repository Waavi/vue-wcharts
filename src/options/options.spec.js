import {
    theme, colors, setTheme, setOptions, setColors,
} from '.'

describe('Options/colors', () => {
    const newColors = ['#fff', '#000']

    xit('Set theme colors via setColors correctly', () => {
        setColors(newColors)
        expect(colors).toEqual(newColors)
    })
    xit('Set theme colors via setOptions correctly', () => {
        setOptions({ colors: newColors })
        expect(colors).toEqual(newColors)
    })
})
describe('Options/theme', () => {
    const newTheme = {
        WLine: {
            styles: {
                fill: '#fff',
                stroke: '#000',
                strokeWidth: 4,
                strokeDasharray: 1,
            },
        },
    }

    xit('Set theme via setTheme correctly', () => {
        setTheme(newTheme)
        expect(theme).toMatchSnapshot()
    })
    xit('Set theme theme via setOptions correctly', () => {
        setOptions({ theme: newTheme })
        expect(theme).toMatchSnapshot()
    })
})
