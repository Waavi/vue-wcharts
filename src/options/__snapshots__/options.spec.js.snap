// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options/theme Set theme theme via setOptions correctly 1`] = `
Object {
  "WAxisText": Object {
    "styles": Object {
      "fill": "none",
      "fontSize": 14,
      "stroke": "none",
    },
  },
  "WBar": Object {
    "background": Object {
      "fill": "#eeeeee",
    },
    "label": Object {
      "cursor": "default",
      "fill": "#333333",
    },
    "stackedLabel": Object {
      "cursor": "default",
      "fill": "#333333",
    },
    "styles": Object {},
  },
  "WBullet": Object {
    "styles": Object {
      "background": "white",
      "borderColor": "white",
      "borderRadius": "50%",
      "borderStyle": "solid",
      "borderWidth": "2px",
      "display": "inline-block",
      "height": "8px",
      "margin": "0px 8px 0px 0px",
      "width": "8px",
    },
  },
  "WCartesianGrid": Object {
    "styles": Object {
      "stroke": "#cccccc",
      "strokeDasharray": "3",
      "strokeWidth": 1,
    },
  },
  "WDot": Object {
    "hoverRadius": 8,
    "radius": 4,
    "strokeWidth": 2,
    "styles": Object {
      "cursor": "default",
    },
  },
  "WLegend": Object {
    "bulletStyles": Object {
      "background": "white",
      "borderColor": "white",
      "borderRadius": "50%",
      "borderStyle": "solid",
      "borderWidth": "2px",
      "display": "inline-block",
      "height": "8px",
      "margin": "0px 8px 0px 0px",
      "width": "8px",
    },
    "legendStyles": Object {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "fontSize": "12px",
    },
    "legendStylesDisabled": Object {
      "cursor": "default",
      "opacity": "0.5",
    },
    "styles": Object {
      "display": "flex",
      "position": "absolute",
    },
    "wrapperStyles": Object {
      "display": "flex",
      "flex": "1",
      "listStyle": "none",
      "margin": "0",
    },
  },
  "WLegendItem": Object {
    "bulletStyles": Object {
      "background": "white",
      "borderColor": "white",
      "borderRadius": "50%",
      "borderStyle": "solid",
      "borderWidth": "2px",
      "display": "inline-block",
      "height": "8px",
      "margin": "0px 8px 0px 0px",
      "width": "8px",
    },
    "disabledStyles": Object {
      "cursor": "default",
      "opacity": "0.5",
    },
    "styles": Object {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "fontSize": "12px",
    },
  },
  "WLine": Object {
    "dot": Object {
      "fill": "white",
      "hoverRadius": 8,
      "radius": 4,
      "stroke": "",
      "strokeWidth": 2,
    },
    "styles": Object {
      "fill": "#fff",
      "stroke": "#000",
      "strokeDasharray": 1,
      "strokeWidth": 4,
    },
  },
  "WMarker": Object {
    "label": Object {
      "fill": "#333333",
      "fontSize": 14,
    },
    "styles": Object {
      "fill": "none",
      "r": 10,
      "stroke": "#333333",
      "strokeDasharray": "0",
      "strokeWidth": 1,
    },
  },
  "WPie": Object {
    "path": Object {
      "stroke": "#ffffff",
    },
    "styles": Object {
      "position": "relative",
      "transform": "translate(50%, 50%)",
    },
  },
  "WScatter": Object {
    "dot": Object {
      "fill": "",
      "opacity": 0.8,
      "radius": 8,
      "stroke": "",
      "strokeWidth": 0,
    },
    "line": Object {
      "fill": "",
      "stroke": "",
      "strokeDasharray": "0",
      "strokeWidth": 1,
    },
  },
  "WSimpleHStackBar": Object {
    "label": Object {
      "color": "#ffffff",
      "left": "5px",
      "position": "absolute",
      "top": "50%",
      "transform": "translateY(-50%)",
    },
    "marker": Object {
      "background": "#ffffff",
      "height": "10px",
      "position": "absolute",
      "top": "50%",
      "transform": "translateY(-50%)",
      "width": "10px",
    },
    "styles": Object {
      "height": "20px",
    },
  },
  "WTooltip": Object {
    "styles": Object {
      "background": "white",
      "borderRadius": "4px",
      "boxShadow": "0 2px 10px 0 rgba(0, 0, 0, 0.15)",
      "color": "#000000",
      "display": "flex",
      "fontSize": "14px",
      "opacity": 0,
      "padding": ".5rem",
      "position": "absolute",
      "transition": "opacity .25s ease",
      "zIndex": 1,
    },
    "title": Object {
      "marginBottom": "5px",
    },
    "visible": Object {
      "opacity": 1,
    },
    "wrapper": Object {
      "display": "flex",
      "flexDirection": "column",
    },
  },
  "WXAxis": Object {
    "label": Object {
      "fill": "#414141",
      "fontWeight": 800,
    },
    "mark": Object {
      "stroke": "#cccccc",
    },
    "styles": Object {
      "stroke": "#cccccc",
    },
    "tick": Object {
      "fill": "#414141",
      "fontSize": 12,
      "stroke": "none",
    },
  },
  "WYAxis": Object {
    "label": Object {
      "fill": "#414141",
      "fontWeight": 800,
    },
    "mark": Object {
      "stroke": "#cccccc",
    },
    "styles": Object {
      "stroke": "#cccccc",
    },
    "tick": Object {
      "fill": "#414141",
      "fontSize": 12,
      "stroke": "none",
    },
  },
  "WZAxis": Object {
    "label": Object {
      "fill": "#414141",
      "fontWeight": 800,
    },
    "mark": Object {
      "stroke": "#cccccc",
    },
    "styles": Object {
      "stroke": "#cccccc",
    },
    "tick": Object {
      "fill": "#414141",
      "fontSize": 12,
      "stroke": "none",
    },
  },
}
`;

exports[`Options/theme Set theme via setTheme correctly 1`] = `
Object {
  "WAxisText": Object {
    "styles": Object {
      "fill": "none",
      "fontSize": 14,
      "stroke": "none",
    },
  },
  "WBar": Object {
    "background": Object {
      "fill": "#eeeeee",
    },
    "label": Object {
      "cursor": "default",
      "fill": "#333333",
    },
    "stackedLabel": Object {
      "cursor": "default",
      "fill": "#333333",
    },
    "styles": Object {},
  },
  "WBullet": Object {
    "styles": Object {
      "background": "white",
      "borderColor": "white",
      "borderRadius": "50%",
      "borderStyle": "solid",
      "borderWidth": "2px",
      "display": "inline-block",
      "height": "8px",
      "margin": "0px 8px 0px 0px",
      "width": "8px",
    },
  },
  "WCartesianGrid": Object {
    "styles": Object {
      "stroke": "#cccccc",
      "strokeDasharray": "3",
      "strokeWidth": 1,
    },
  },
  "WDot": Object {
    "hoverRadius": 8,
    "radius": 4,
    "strokeWidth": 2,
    "styles": Object {
      "cursor": "default",
    },
  },
  "WLegend": Object {
    "bulletStyles": Object {
      "background": "white",
      "borderColor": "white",
      "borderRadius": "50%",
      "borderStyle": "solid",
      "borderWidth": "2px",
      "display": "inline-block",
      "height": "8px",
      "margin": "0px 8px 0px 0px",
      "width": "8px",
    },
    "legendStyles": Object {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "fontSize": "12px",
    },
    "legendStylesDisabled": Object {
      "cursor": "default",
      "opacity": "0.5",
    },
    "styles": Object {
      "display": "flex",
      "position": "absolute",
    },
    "wrapperStyles": Object {
      "display": "flex",
      "flex": "1",
      "listStyle": "none",
      "margin": "0",
    },
  },
  "WLegendItem": Object {
    "bulletStyles": Object {
      "background": "white",
      "borderColor": "white",
      "borderRadius": "50%",
      "borderStyle": "solid",
      "borderWidth": "2px",
      "display": "inline-block",
      "height": "8px",
      "margin": "0px 8px 0px 0px",
      "width": "8px",
    },
    "disabledStyles": Object {
      "cursor": "default",
      "opacity": "0.5",
    },
    "styles": Object {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "fontSize": "12px",
    },
  },
  "WLine": Object {
    "dot": Object {
      "fill": "white",
      "hoverRadius": 8,
      "radius": 4,
      "stroke": "",
      "strokeWidth": 2,
    },
    "styles": Object {
      "fill": "#fff",
      "stroke": "#000",
      "strokeDasharray": 1,
      "strokeWidth": 4,
    },
  },
  "WMarker": Object {
    "label": Object {
      "fill": "#333333",
      "fontSize": 14,
    },
    "styles": Object {
      "fill": "none",
      "r": 10,
      "stroke": "#333333",
      "strokeDasharray": "0",
      "strokeWidth": 1,
    },
  },
  "WPie": Object {
    "path": Object {
      "stroke": "#ffffff",
    },
    "styles": Object {
      "position": "relative",
      "transform": "translate(50%, 50%)",
    },
  },
  "WScatter": Object {
    "dot": Object {
      "fill": "",
      "opacity": 0.8,
      "radius": 8,
      "stroke": "",
      "strokeWidth": 0,
    },
    "line": Object {
      "fill": "",
      "stroke": "",
      "strokeDasharray": "0",
      "strokeWidth": 1,
    },
  },
  "WSimpleHStackBar": Object {
    "label": Object {
      "color": "#ffffff",
      "left": "5px",
      "position": "absolute",
      "top": "50%",
      "transform": "translateY(-50%)",
    },
    "marker": Object {
      "background": "#ffffff",
      "height": "10px",
      "position": "absolute",
      "top": "50%",
      "transform": "translateY(-50%)",
      "width": "10px",
    },
    "styles": Object {
      "height": "20px",
    },
  },
  "WTooltip": Object {
    "styles": Object {
      "background": "white",
      "borderRadius": "4px",
      "boxShadow": "0 2px 10px 0 rgba(0, 0, 0, 0.15)",
      "color": "#000000",
      "display": "flex",
      "fontSize": "14px",
      "opacity": 0,
      "padding": ".5rem",
      "position": "absolute",
      "transition": "opacity .25s ease",
      "zIndex": 1,
    },
    "title": Object {
      "marginBottom": "5px",
    },
    "visible": Object {
      "opacity": 1,
    },
    "wrapper": Object {
      "display": "flex",
      "flexDirection": "column",
    },
  },
  "WXAxis": Object {
    "label": Object {
      "fill": "#414141",
      "fontWeight": 800,
    },
    "mark": Object {
      "stroke": "#cccccc",
    },
    "styles": Object {
      "stroke": "#cccccc",
    },
    "tick": Object {
      "fill": "#414141",
      "fontSize": 12,
      "stroke": "none",
    },
  },
  "WYAxis": Object {
    "label": Object {
      "fill": "#414141",
      "fontWeight": 800,
    },
    "mark": Object {
      "stroke": "#cccccc",
    },
    "styles": Object {
      "stroke": "#cccccc",
    },
    "tick": Object {
      "fill": "#414141",
      "fontSize": 12,
      "stroke": "none",
    },
  },
  "WZAxis": Object {
    "label": Object {
      "fill": "#414141",
      "fontWeight": 800,
    },
    "mark": Object {
      "stroke": "#cccccc",
    },
    "styles": Object {
      "stroke": "#cccccc",
    },
    "tick": Object {
      "fill": "#414141",
      "fontSize": 12,
      "stroke": "none",
    },
  },
}
`;
